src/tools/optimization-options.h

```diff
@@ -18,67 +18,116 @@
-             runOptimizationPasses = true;
-           })
-      .add("", "-O0", "execute no optimization passes",
-           Options::Arguments::Zero,
-           [&passOptions](Options*, const std::string&) {
-             passOptions.optimizeLevel = 0;
-             passOptions.shrinkLevel = 0;
-           })
-      .add("", "-O1", "execute -O1 optimization passes",
-           Options::Arguments::Zero,
-           [&runOptimizationPasses, &passOptions](Options*, const std::string&) {
-             passOptions.optimizeLevel = 1;
-             passOptions.shrinkLevel = 0;
-             runOptimizationPasses = true;
-           })
-      .add("", "-O2", "execute -O2 optimization passes",
-           Options::Arguments::Zero,
-           [&runOptimizationPasses, &passOptions](Options*, const std::string&) {
-             passOptions.optimizeLevel = 2;
-             passOptions.shrinkLevel = 0;
-             runOptimizationPasses = true;
-           })
-      .add("", "-O3", "execute -O3 optimization passes",
-           Options::Arguments::Zero,
-           [&runOptimizationPasses, &passOptions](Options*, const std::string&) {
-             passOptions.optimizeLevel = 3;
-             passOptions.shrinkLevel = 0;
-             runOptimizationPasses = true;
-           })
-      .add("", "-Os", "execute default optimization passes, focusing on code size",
-           Options::Arguments::Zero,
-           [&runOptimizationPasses, &passOptions](Options*, const std::string&) {
-             passOptions.optimizeLevel = 2;
-             passOptions.shrinkLevel = 1;
-             runOptimizationPasses = true;
-           })
-      .add("", "-Oz", "execute default optimization passes, super-focusing on code size",
-           Options::Arguments::Zero,
-           [&runOptimizationPasses, &passOptions](Options*, const std::string&) {
-             passOptions.optimizeLevel = 2;
-             passOptions.shrinkLevel = 2;
-             runOptimizationPasses = true;
-           })
-      .add("--optimize-level", "-ol", "How much to focus on optimizing code",
-           Options::Arguments::One,
-           [&passOptions](Options* o, const std::string& argument) {
-             passOptions.optimizeLevel = atoi(argument.c_str());
-           })
-      .add("--shrink-level", "-s", "How much to focus on shrinking code size",
-           Options::Arguments::One,
-           [&passOptions](Options* o, const std::string& argument) {
-             passOptions.shrinkLevel = atoi(argument.c_str());
-           })
-      .add("--ignore-implicit-traps", "-iit", "Optimize under the helpful assumption that no surprising traps occur (from load, div/mod, etc.)",
-           Options::Arguments::Zero,
-           [&passOptions](Options*, const std::string&) {
-             passOptions.ignoreImplicitTraps = true;
-           })
+namespace wasm {
+
+struct OptimizationOptions : public Options {
+  const std::string DEFAULT_OPT_PASSES = "O";
```

Comment:
Hmm, that does sound like a good idea, but when I try it I get
````
in-class initialization of static data member ‘const string wasm::OptimizationOptions::DEFAULT_OPT_PASSES’ of non-literal type

call to non-constexpr function
````
I tried changing it to a `const char*` but it still complains about
````
error: ‘constexpr’ needed for in-class initialization of static data member ‘const char* wasm::OptimizationOptions::DEFAULT_OPT_PASSES’ of non-integral type
````